From 3840a0b9fd24de7a2b2cc87c96e79ff00cd324d5 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:43:15 -0400
Subject: [ethernet] cxgb4: use the correct max size for firmware flash

Message-id: <1410475447-94392-57-git-send-email-svemuri@redhat.com>
Patchwork-id: 93849
O-Subject: [RHEL7.1 PATCH BZ 1124947 056/108] cxgb4: use the correct max size for firmware flash
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

The wrong max fw size was being used and causing false
"too big" errors running ethtool -f.

Signed-off-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 6f1d7210376727d090e04b8635e6dda4d7eb7b0c)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
index 66004ac..d07c2ad 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
@@ -683,7 +683,7 @@ enum {
  SF_RD_ID        = 0x9f,       /* read ID */
  SF_ERASE_SECTOR = 0xd8,       /* erase sector */
 
- FW_MAX_SIZE = 512 * 1024,
+ FW_MAX_SIZE = 16 * SF_SEC_SIZE,
 };
 
 /**
-- 
1.7.1